functions-app/report-psychometrician/service/data-service/psychometrician-report-cache.data.service.js

```diff
@@ -49,19 +60,18 @@ const psychometricianReportCacheDataService = {
     if (!batchSize) {
       throw new Error('Missing argument: batchSize')
     }
-    const safeBatchSize = parseInt(batchSize, 10)
-
-    const sql = `SELECT TOP ${safeBatchSize} chk.id
-      FROM [mtc_admin].[check] chk
-      LEFT JOIN [mtc_admin].psychometricianReportCache prc ON (chk.id = prc.check_id)
-      JOIN [mtc_admin].[checkStatus] cs ON (chk.checkStatus_id = cs.id)
-      WHERE
-        prc.check_id IS NULL
-      AND ((cs.code = 'CMP' AND chk.markedAt IS NOT NULL) OR cs.code = 'NTR')
-      ORDER BY NEWID()`
-
+    const sql = getUnprocessedChecksSql(batchSize)
     const results = await sqlService.query(sql)
     return results.map(r => r.id)
+  },
+
+  hasUnprocessedChecks: async function () {
+    const sql = getUnprocessedChecksSql(1)
+    const results = await sqlService.query(sql)
+    if (results && results.length) {
```

Comment:
@jon-shipley can length ever be 0 here? shouldn't you also be checking if length is > 0?

Reply:
Yes, it can be 0, and, no, I don't think it needs to be `length > 0`.  Shorter seems clearer to me.

Reply:
wouldn't 0 make your expected result wrong?